File: PASCAL/Cap07/c07ex06.pas
program c07ex06;

function FATORIAL(N: Integer): Integer;
var
  FAT, I: Integer;
begin
  FAT := 1;
  for I := 1 to N do
    begin
      FAT := FAT * I;
    end;
  FATORIAL := FAT;
end;

var
  LIMITE: Integer;

begin
  Write('Qual fatorial: ');
  Readln(LIMITE);

  Writeln('Fatorial = ', FATORIAL(LIMITE));

  Writeln;
  Write('Tecle <Enter> para encerrar... ');
  Readln;
end.

File: LUA/Cap07/c07ex06.lua
-- c07ex06.lua

function fatorial(n)
  local fat = 1
  for i = 1, n do
    fat = fat * i
  end
  return fat
end

io.write("Qual fatorial: ")
local limite = tonumber(io.read())

print("Fatorial = " .. fatorial(limite))

io.write("\nTecle <Enter> para encerrar... ")
io.read()
